garrysmod/lua/includes/modules/undo.lua

```diff
@@ -356,7 +356,9 @@ function Do_Undo( undo )
 
 	if ( !undo ) then return false end
 	
-	hook.Run( "PreUndo", undo )
+	local call = hook.Run( "PreUndo", undo )
+
+	if ( !call ) then return end

```

Comment:
This should be ``if ( call == false ) then return end``, othewise, all undos would be blocked. You should also add default GM funcs for PreUndo and PostUndo.

Reply:
@Kefta What do you mean by default GM funcs?

Reply:
> @Kefta What do you mean by default GM funcs?

Add GM:PreUndo (returning true) and GM:PostUndo to the base gamemode.